File: app/src/main/java/org/digitalcampus/oppia/utils/storage/ExternalStorageState.java
package org.digitalcampus.oppia.utils.storage;

import android.os.Environment;

import java.io.File;

public class ExternalStorageState {

    public static final String STATE_NOT_WRITABLE = "STATE_NOT_WRITABLE";

    private static ExternalStorageState externalStorageState = new ExternalStorageState();

    public static String getExternalStorageState(File file){ return externalStorageState.getState(file);  }

    protected String getState(File file) { return Environment.getExternalStorageState(file);}

    public static void setExternalStorageState(ExternalStorageState state){
        externalStorageState = state;
    }
}

File: app/src/main/java/org/digitalcampus/oppia/utils/storage/ExternalStorageState.kt
package org.digitalcampus.oppia.utils.storage

import android.os.Environment
import org.digitalcampus.oppia.annotations.Mockable
import java.io.File

@Mockable
class ExternalStorageState {
    fun getState(file: File?): String {
        return Environment.getExternalStorageState(file)
    }

    companion object {
        const val STATE_NOT_WRITABLE = "STATE_NOT_WRITABLE"
        private var externalStorageState = ExternalStorageState()

        @JvmStatic
        fun getExternalStorageState(file: File): String {
            return externalStorageState.getState(file)
        }

        @JvmStatic
        fun setExternalStorageState(state: ExternalStorageState) {
            externalStorageState = state
        }
    }
}